app/src/data/useDatasetStore.ts

```diff
@@ -194,21 +193,34 @@ export function useDatasetStoreProvider(): DatasetStore {
         // you request covid cases we could also cache it under covid deaths
         // since they're provided together. Also, it would be nice to cache ACS
         // when it's used from within another provider.
-        const variables = providers.map((provider) =>
-          provider.getData(datasetMap, query.breakdowns)
-        );
-        const dataframes: IDataFrame[] = variables.map(
-          (data) => new DataFrame(data)
-        );
-        const joined = dataframes.reduce((prev, next) => {
-          return joinOnCols(
-            prev,
-            next,
-            query.breakdowns.getJoinColumns(),
-            query.joinType
+        try {
+          const variables = providers.map((provider) =>
+            provider.getData(datasetMap, query.breakdowns)
           );
-        });
-        return joined.toArray();
+
+          const dataframes: IDataFrame[] = variables.map(
+            (data) => new DataFrame(data)
+          );
+
+          const joined = dataframes.reduce((prev, next) => {
+            return joinOnCols(
+              prev,
+              next,
+              query.breakdowns.getJoinColumns(),
+              query.joinType
+            );
+          });
+          return new MetricQueryResponse(joined.toArray());
+        } catch (err) {
+          if (
+            err instanceof UnsupportedBreakdownError ||
+            err instanceof NoDataError
+          ) {
+            return new MetricQueryResponse(err);
+          } else {
+            throw err; // re-throw the error unchanged
+          }
```

Comment:
Couple thoughts on error handling here:
- did you intentionally leave out MissingDatasetError or was that supposed to be part of the if statement?
- one thing to note is that if you catch the error and convert it to a MetricQueryResponse, the error won't be logged. If the error is not really an error but an expected scenario, then that's fine. Otherwise, we should log it here.
- Generally, I'm not a huge fan of throwing errors for expected cases, as I think it blurs lines like when to recover vs when to let the error propagate, when to log errors, when to show specifics in the UI vs generic error, etc. I would recommend using return values instead of errors for expected cases, and let all errors propagate here. In this case, "unsupported breakdown" is expected, "missing dataset" is unexpected, and I think "no data" is unexpected but I'm not 100% sure on that one.

Reply:
Bug filed per our discussion